### ISSUE TITLE
Tiles missing data when multiple rows have same id

### ISSUE THREAD

**Issue by pnorman** [go-spatial/tegola#337]
Some tiles are empty with no apparent reason. Although only one rendered tile is visible here, others are scattered around.

![image](https://user-images.githubusercontent.com/1190866/36648631-b77e8d70-1a4a-11e8-9fef-a34f09a4a7f7.png)

In an effort to debug I ran with `TEGOLA_OPTIONS=DontSimplifyGeo SQL_DEBUG=EXECUTE_SQL` and had no changes.

After a bit of debugging, I looked at the geojson, and it's odd. I got this, where translucent green is polygons in the tile

![image](https://user-images.githubusercontent.com/1190866/36648965-3da85c48-1a4e-11e8-803b-42be09610de9.png)

The lower-left is the only data from the water layer.

![image](https://user-images.githubusercontent.com/1190866/36648846-3dd42b58-1a4d-11e8-8ebd-a3b835f0f613.png)

There's something odd going on when I zoom in close, with several things happening

![image](https://user-images.githubusercontent.com/1190866/36648914-d2c8d13c-1a4d-11e8-8130-b4885457ea7d.png)

The purple polygons overlap on the tile edge. This is intentional, but makes it look more complex.

Red dots are on vertexes of the original data, but you can see the vector tile has introduced additional ones. Coordinates look slightly off but that could be because it's zoomed in so much the vector tile grid resolution matters.

Shell commands for converting were
```sh
curl 'http://192.168.1.1:8080/maps/bolder/7/19/43.pbf?debug=true' -o 7_19_43.pbf
vt2geojson -z 7 -x 19 -y 43 7_19_43.pbf > 7_19_43.geojson
```

**Comment by pnorman** [comment]
Oh, for SQL-side debugging, I checked that particular tile with 

```sql
select osm_id, st_area(way::geometry) from (SELECT
      0::bigint AS osm_id,
      ST_AsBinary(way) AS way
    FROM simplified_ocean_polygons
    WHERE way && ST_MakeEnvelope(-1.4093765021372069e+07,6.57969939387207e+06,-1.3770895013940427e+07,6.25682938644043e+06,3857)) as _;
```

Results were
```
┌────────┬──────────────────┐
│ osm_id │     st_area      │
├────────┼──────────────────┤
│      0 │ 359223880333.241 │
│      0 │ 15631132.2569061 │
│      0 │ 377659.520064561 │
│      0 │ 5031.66775602203 │
│      0 │ 632656537.448886 │
│      0 │ 53202350.4275055 │
│      0 │ 155817611103.982 │
│      0 │ 17527459038.9674 │
│      0 │ 607717.938907239 │
│      0 │ 856719.443274425 │
│      0 │ 6386050769.74848 │
│      0 │ 60422364.8476324 │
│      0 │ 13837482.8106902 │
│      0 │ 1479355.45746988 │
└────────┴──────────────────┘
```

For scale, at traditional z7 pixel is 1490116 square Mercator meters, and a 4096 grid square is 5776. Several of the rows there are of significant size.

**Comment by ARolek** [comment]
@pnorman as always, thanks for the detailed submission. I'm wondering if the fact that all the features have the same `osm_id` might be causing the issue? I believe that the feature ID needs to be unique or it can cause problems. If this is the case, then we need to add a log statement when this occurs.

**Comment by pnorman** [comment]
> I believe that the feature ID needs to be unique or it can cause problems.

Having a unique ID is only a SHOULD in the spec, it's not required at all.

**Comment by ARolek** [comment]
@pnorman this has been fixed on the `v0.8.x` branch.

**Comment by pnorman** [comment]
Confirming fixed.
